launch_testing/launch_testing/__init__.py

```diff
@@ -70,6 +77,9 @@ def on_stdout_lines(self, lines):
 
         for line in lines.splitlines():
             # Filter out stdout that comes from underlying DDS implementation
+            # Note: we do not currently support matching filters across multiple stdout lines.
+            if any(re.fullmatch(pattern, line) for pattern in self.filtered_patterns):
```

Comment:
Does this need to be a `fullmatch`? Could it just use `match` instead and leave it up to the regex to use `^` and `$`?

Reply:
I think I was just trying to avoid regexes from matching accidentally, but I agree that that should be the choice of the user.

If we just use `match`, which only matches at the start of a line, then actually the prefix option could then be collapsed back into the pattern option... in which case we can just document that filters can be literal strings of prefixes for those users just wanting the simplicity of specifying prefixes. How's that sound?

Reply:
I don't think you can determine if the filter is supposed to be a string or a regular expression. E.g. if it contains a `.` you can't guess the semantic. Therefore I think you need to know the type of the filter (prefix string vs. regex) in order to invoke the corresponding logic.

Again, I think we should never use a regular expression match when a string match is sufficient  simply because of the unjustified overhead.